### PR TITLE
Hash KubeadmControlPlane.Spec

### PR DESCRIPTION
**Pull request by mnitchev**
### Checklist

- [ ] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create

### CODE REVIEW

File: helm/cluster-gcp/templates/_control_plane.tpl [hunk 1]
```diff
@@ -20,14 +20,128 @@ serviceAccounts:
...
+
 
 {{- define "control-plane" }}
 apiVersion: controlplane.cluster.x-k8s.io/v1beta1
 kind: KubeadmControlPlane
 metadata:
   labels:
     {{- include "labels.common" $ | nindent 4 }}
-  name: {{ include "resource.default.name" $ }}
+  name: {{ include "resource.default.name" $ }}-{{ include "hash" (dict "data" (include "controlplane-kubeadmcontrolplane-spec" $) "global" .) }}
```
Comment (AverageMarcus) [inline helm/cluster-gcp/templates/_control_plane.tpl:144]: You need to update the name here too: https://github.com/giantswarm/cluster-gcp/blob/main/helm/cluster-gcp/templates/_cluster.tpl#L19

**Review by AverageMarcus — CHANGES_REQUESTED** [review]

File: helm/cluster-gcp/templates/_control_plane.tpl (hunk 1 above)
Comment (mnitchev) [reply helm/cluster-gcp/templates/_control_plane.tpl:144]: done, thanks!

**Review by AverageMarcus — APPROVED** [review]

**Review by zewolfe — APPROVED** [review]

**Comment by mnitchev** [comment]
Trying this out on some WCs it doesn't look like there's a safe way to get this to work. Replacing the KubeadmControlPlane will instantly cause the control-plane machines to be killed, after which reconciliation is impossible. Since this change anyway cannot be applied (the initConfiguration is only run on the first control-plane node that joins) and the upgrade needs to be done manually, I'm going to go ahead and close this PR.

**Pull request closed without merging**